### ISSUE TITLE
bogus escape: '\\x' on word documents identification (container-signature updated : container-signature-20140923.xml)

### ISSUE THREAD

**Issue by florianajir** [openpreserve/fido#56]
I tried to update conf/container-signature file (container-signature-20140923.xml) to see the difference on word document identification and the result is bad : docx files are no more recognized (identify as a zip), see : 

with container-signature-20140923.xml :
```
FIDO v1.3.1 (formats-v78.xml, container-signature-20140923.xml, format_extensions.xml)
OK,240,fmt/40,"Microsoft Word for Windows Document","Microsoft Word for Windows 97 - 2002",11776,"/home/fajir/docs/AnnexeDoc.doc","application/msword","signature"
bogus escape: '\\x'
OK,10,x-fmt/263,"ZIP Format","ZIP format",25204,"/home/fajir/docs/cours sur les theories de la motivation.docx","application/zip","signature"
```

with container-signature-20130501.xml :
```
FIDO v1.3.1 (formats-v78.xml, container-signature-20130501.xml, format_extensions.xml)
OK,230,fmt/40,"Microsoft Word for Windows Document","Microsoft Word for Windows 97 - 2002",11776,"/home/fajir/docs/AnnexeDoc.doc","application/msword","signature"
OK,29,fmt/412,"Microsoft Office Open XML - Word","Microsoft Office Open XML - Word",25204,"/home/fajir/docs/cours sur les theories de la motivation.docx","None","signature"
```

I think the bug come from "bogus escape: '\\x'"

**Comment by mistydemeo** [comment]
I've fixed the regex generation in #59, however this doesn't fix the misidentification (which I am also seeing in some Office files). I can identify the same files using Siegfried correctly, so I believe the problem is in FIDO and not the signature files.

**Comment by mistydemeo** [comment]
DROID also returns a correct match using up-to-date container signatures.

**Comment by mistydemeo** [comment]
In my case, the file is an Excel document. The matching container signature should be one of the following:

```xml
        <ContainerSignature Id="2040" ContainerType="ZIP">
            <Description>Microsoft Excel OOXML (little endian)</Description>
            <Files>
                <File>
                    <Path>[Content_Types].xml</Path>
                    <BinarySignatures>
                        <InternalSignatureCollection>
	                        <InternalSignature ID="317">
	                            <ByteSequence Reference="BOFoffset">
	                                <SubSequence Position="1" SubSeqMinOffset="0" SubSeqMaxOffset="32768">
	                                    <Sequence>'C'00 'o'00 'n'00 't'00 'e'00 'n'00 't'00 'T'00 'y'00 'p'00 'e'00 '='00 '"'00 'a'00 'p'00 'p'00 'l'00 'i'00 'c'00 'a'00 't'00 'i'00 'o'00 'n'00 '/'00 'v'00 'n'00 'd'00 '.'00 'o'00 'p'00 'e'00 'n'00 'x'00 'm'00 'l'00 'f'00 'o'00 'r'00 'm'00 'a'00 't'00 's'00 '-'00 'o'00 'f'00 'f'00 'i'00 'c'00 'e'00 'd'00 'o'00 'c'00 'u'00 'm'00 'e'00 'n'00 't'00 '.'00 's'00 'p'00 'r'00 'e'00 'a'00 'd'00 's'00 'h'00 'e'00 'e'00 't'00 'm'00 'l'00 '.'00 's'00 'h'00 'e'00 'e'00 't'00 '.'00 'm'00 'a'00 'i'00 'n'00 '+'00 'x'00 'm'00 'l'00 '"'00</Sequence>
	                                </SubSequence>
	                            </ByteSequence>
	                        </InternalSignature>
                        </InternalSignatureCollection>
                    </BinarySignatures>
                </File>
            </Files>
        </ContainerSignature> 

        <ContainerSignature Id="2050" ContainerType="ZIP">
            <Description>Microsoft Excel OOXML (big endian)</Description>
            <Files>
                <File>
                    <Path>[Content_Types].xml</Path>
                    <BinarySignatures>
                        <InternalSignatureCollection>
	                        <InternalSignature ID="317">
	                            <ByteSequence Reference="BOFoffset">
	                                <SubSequence Position="1" SubSeqMinOffset="0" SubSeqMaxOffset="32768">
	                                    <Sequence>00'C' 00'o' 00'n' 00't' 00'e' 00'n' 00't' 00'T' 00'y' 00'p' 00'e' 00'=' 00'"' 00'a' 00'p' 00'p' 00'l' 00'i' 00'c' 00'a' 00't' 00'i' 00'o' 00'n' 00'/' 00'v' 00'n' 00'd' 00'.' 00'o' 00'p' 00'e' 00'n' 00'x' 00'm' 00'l' 00'f' 00'o' 00'r' 00'm' 00'a' 00't' 00's' 00'-' 00'o' 00'f' 00'f' 00'i' 00'c' 00'e' 00'd' 00'o' 00'c' 00'u' 00'm' 00'e' 00'n' 00't' 00'.' 00's' 00'p' 00'r' 00'e' 00'a' 00'd' 00's' 00'h' 00'e' 00'e' 00't' 00'm' 00'l' 00'.' 00's' 00'h' 00'e' 00'e' 00't' 00'.' 00'm' 00'a' 00'i' 00'n' 00'+' 00'x' 00'm' 00'l' 00'"'</Sequence>
	                                </SubSequence>
	                            </ByteSequence>
	                        </InternalSignature>
                        </InternalSignatureCollection>
                    </BinarySignatures>
                </File>
            </Files>
        </ContainerSignature> 
```

However, the regex produced by FIDO does not match either file.

**Comment by mistydemeo** [comment]
Deleted my previous comment, which was incorrect.

In my case, I believe the matching signature is this one:

```xml
        <ContainerSignature Id="2030" ContainerType="ZIP">
            <Description>Microsoft Excel OOXML</Description>
            <Files>
                <File>
                    <Path>[Content_Types].xml</Path>
                    <BinarySignatures>
                        <InternalSignatureCollection>
	                        <InternalSignature ID="317">
	                            <ByteSequence Reference="BOFoffset">
	                                <SubSequence Position="1" SubSeqMinOffset="0" SubSeqMaxOffset="32768">
	                                    <Sequence>'ContentType="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet.main+xml"'</Sequence>
	                                </SubSequence>
	                            </ByteSequence>
	                        </InternalSignature>
                        </InternalSignatureCollection>
                    </BinarySignatures>
                </File>
            </Files>
        </ContainerSignature> 
```

FIDO produces the following regex from that signature:

```
(?s)ContentType="application/vnd\.openxmlformats-officedocument\.spreadsheetml\.sheet\.main\+xml"
```

Which *is* a valid regex which should match the file in question.

**Comment by mistydemeo** [comment]
Oh, I think I see what's going on. FIDO doesn't appear to do special handling of ZIP containers, e.g. it doesn't decompress them before attempting to read their contents, which means it can't match the plaintext sequences in the container signatures.

**Comment by mistydemeo** [comment]
I've added support for ZIP containers; there's a pull request open, #60.

**Comment by Hwesta** [comment]
@florianajir Misty's PR #60 was merged and is part of 1.3.4 - are you still seeing this issue?

**Comment by florianajir** [comment]
All right I close the issue, thanks
